include/nuttx/can/can.h

```diff
@@ -246,6 +257,46 @@
  *                   is returned with the errno variable set to indicate the
  *                   nature of the error.
  *   Dependencies:   None
+ *
+ * CANIOC_SET_STATE
+ *   Description:    Set specfic can controller state
+ *
+ *   Argument:       A pointer to an int type that describes the CAN
+ *                   controller state.
+ *   returned Value: Zero (OK) is returned on success.  Otherwise -1 (ERROR)
+ *                   is returned with the errno variable set to indicate the
+ *                   nature of the error.
+ *   Dependencies:   None
+ *
+ * CANIOC_GET_STATE
+ *   Description:    Get specfic can controller state
+ *
+ *   Argument:       A pointer to an int type that describes the CAN
```

Comment:
We cannot avoid pointer here, but maybe the state can be passed as a returned value. I am not convinced which is better for NuttX. We used the returned value to obtain the controller state/info in RTEMS CAN stack implementation.

Reply:
I refer to `CANIOC_GET_CONNMODES`， this cmd use A pointer to a write-able instance of struct canioc_connmodes_s in which the new bus modes will be returned.